File: Lab2.cpp/Vector.cpp
#include "Vector.h"
#include <cmath>
#include <iostream>

Vector::Vector() : x(0), y(0) {}

Vector::Vector(double x, double y) : x(x), y(y) {}

Vector::Vector(const Vector& other) : x(other.x), y(other.y) {}

Vector::~Vector() {
    std::cout << "Vector object is being deleted" << std::endl;
}

double Vector::length() const {
    return std::sqrt(x * x + y * y);
}

void Vector::getPolarCoordinates(double& radius, double& angle) const {
    radius = length();
    angle = std::atan2(y, x);
}
void Vector::print() const {
    std::cout << "Vector: (" << x << ", " << y << ")" << std::endl;
}

File: Lab2.cs/Vector.cs
using System;

namespace Lab2
{
    public class Vector
    {
        private readonly double _x;
        private readonly double _y;

        public Vector() : this(0, 0) { }

        public Vector(double x, double y)
        {
            _x = x;
            _y = y;
        }
        ~Vector()
        {
            Console.WriteLine("Vector object is being deleted");
        }
        public double Length()
        {
            return Math.Sqrt(_x * _x + _y * _y);
        }
        public (double Radius, double Angle) GetPolarCoordinates()
        {
            double radius = Length();
            double angle = Math.Atan2(_y, _x);
            return (radius, angle);
        }

        public override string ToString()
        {
            return $"({_x}, {_y})";
        }
    }
}